src/python/pants/reporting/reporting.py

```diff
@@ -112,6 +112,11 @@ def initialize(self, run_tracker, all_options, start_time=None):
       raise ValueError(
         "Flags zipkin-trace-id and zipkin-parent-id must both either be set or not set."
       )
+
+    # If trace_id isn't set by a flag, use UUID from run_id
+    if trace_id is None:
+      _, _, trace_id = run_id.rpartition("_")
```

Comment:
Rather than parsing this out (and assuming what the format of the run_id is), let's have `RunTracker.initialize` return a tuple of `(run_id, run_uuid)` and unpack it in this method